server.js

```diff
@@ -684,6 +684,33 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Coverity Code Advisor On Demand integration
+camp.route(/^\/coverity\/ondemand\/(.+)\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var badgeType = match[1]; // streams or jobs
+  var badgeTypeId = match[2]; // streamId or jobId
+  var format = match[3];
+  var url = 'https://api.ondemand.coverity.com/' + badgeType + '/' + badgeTypeId + '/badge';
+//  var url = 'https://api-stage01.caas.coverity.com/' + badgeType + '/' + badgeTypeId + '/badge';
+//  var url = 'http://localhost:21020/' + badgeType + '/' + badgeTypeId + '/badge';
+  console.log(url)
+  var badgeData = getBadgeData('coverity', data);
+  request(url, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      badgeData = JSON.parse(buffer);
+      sendBadge(format, badgeData);
```

Comment:
Are you sure the format for the JSON we receive is the same as the one we use internally? It is pretty specific.